### PR TITLE
Throw exception if bid and ask are overlapping

### PR DESCRIPTION
**Pull request by joelrich**
See https://github.com/bmoscon/cryptofeed/issues/232 for discussion
This will completely break the Upbit feed if L2_BOOK is enabled (most likely since either the implementation is wrong or the exchange notifications are insufficient).

### CODE REVIEW

File: cryptofeed/feed.py [hunk 1]
```diff
@@ -101,12 +105,22 @@ async def book_callback(self, book: dict, book_type: str, pair: str, forced: boo
...
+        self.check_bid_ask_overlapping(book, pair)
         if book_type == L2_BOOK:
             await self.callback(L2_BOOK, feed=self.id, pair=pair, book=book, timestamp=timestamp, receipt_timestamp=receipt_timestamp)
         else:
             await self.callback(L3_BOOK, feed=self.id, pair=pair, book=book, timestamp=timestamp, receipt_timestamp=receipt_timestamp)
         self.updates[pair] = 0
 
+    def check_bid_ask_overlapping(self, book, pair):
+        bid, ask = book[BID], book[ASK]
+        assert isinstance(bid, sd) and isinstance(ask, sd)
```
Comment (bmoscon) [inline cryptofeed/feed.py:118]: I'm thinking we don't need this as every exchange/feed implementation uses sorted dict.

Comment (joelrich) [reply cryptofeed/feed.py]: Removed

**Comment by bmoscon** [comment]
@joelrich thanks - the build is broken so I'm going to fix that before taking in any more changes. Should be merged shortly

**Comment by bmoscon** [comment]
thanks for the PR. Merged. I added a flag so its optionally enabled when the feed is configured so I can test out different settings on different feeds to try and narrow down a root cause. Could just be lossy exchanges, want to do more digging.

**Comment by joelrich** [comment]
In my log files it has happened for the following exchanges so far:

226 BITTREX   
98 BITFINEX  
77 BITSTAMP  
12 KRAKEN    
7 OKEX     
2 BITMEX

**Pull request merged**